### PR TITLE
chore: upgrades mongodb, fruster and specs

### PR DESCRIPTION
**Pull request by joelso**
Upgrades mongodb versions and fruster deps.

### CODE REVIEW

File: lib/repos/UserRepo.js [hunk 1]
```diff
@@ -184,11 +184,18 @@ class UserRepo {
 			.collation({ locale: config.locale, numericOrdering: true });
 
 
-		let usersFromDatabase = await mongoQueryOperation.skip(start).limit(limit).toArray();
+		if (start) {
```
Comment (dinukafrost) [inline lib/repos/UserRepo.js:187]: Now i got this. this will fail if give 0 value as start.

File: lib/repos/UserRepo.js [hunk 2]
```diff
@@ -277,9 +284,14 @@ class UserRepo {
...
-		const createdUser = await this._collection.insertOne(user);
+		const result = await this._collection.insertOne(user);
 
-		return new UserModel(createdUser.ops[0]);
+		if (result.acknowledged) {
+			const createdUser = await this._collection.findOne({ _id: result.insertedId });
+			return new UserModel(createdUser);
+		} else {
+			throw new Error("Insert failed");
+		}
```
Comment (dinukafrost) [inline lib/repos/UserRepo.js:294]: We are usually not handling exception in the repo. Hope this properly handled in the controller.

File: spec/AddRolesHandler.spec.js [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
-			.testUtilsOptions());
+			.testUtilsOptions(async (conn) => {
+				db = conn.db;
+
+				try {
+					await db.collection(constants.collections.ROLE_SCOPES).deleteMany({});
+					await db.collection(constants.collections.PROFILES).deleteMany({});
+					await db.collection(constants.collections.INITIAL_USER).deleteMany({});
+					await db.collection(constants.collections.USERS).deleteMany({});
+				} catch (e) {}
```
Comment (dinukafrost) [inline spec/AddRolesHandler.spec.js:22]: Why you need to do this? 🤔 I also felt data is not deleting properly. 😕

File: spec/AddSystemRoleHandler.spec.js [hunk 4]
```diff
@@ -52,18 +53,17 @@ describe("AddSystemRoleHandler", () => {
...
+	it("should return 400 if role already exists", async () => {
 		const role = "padmin";
 		const scopes = ["1", "2"];
 
 		await addRole();
 
 		try {
 			await addRole();
-			done.fail();
+			expect(true).toBe(false);
```
Comment (dinukafrost) [inline spec/AddSystemRoleHandler.spec.js]: Why this? 😛 
```suggestion
			fail()
```

File: spec/GetProfilesByQueryHandler.spec.js [hunk 5]
```diff
@@ -9,7 +9,13 @@ describe("GetProfilesByQueryHandler", () => {
 	frusterTestUtils
 		.startBeforeEach({
 			beforeStart: () => config.userFields = constants.dataset.REQUIRED_ONLY,
-			...specConstants.testUtilsOptions()
+			...specConstants.testUtilsOptions(),
+			afterStart: async (connection) => {
+				try {
+					await connection.db.collection(constants.collections.USERS).deleteMany({});
+					console.log()
```
Comment (dinukafrost) [inline spec/GetProfilesByQueryHandler.spec.js]: console.log 💡

**Review by dinukafrost — COMMENTED** [review]
Please check my comments

File: lib/repos/UserRepo.js (hunk 1 above)
Comment (joelso) [reply lib/repos/UserRepo.js:187]: Thinking this will be fine, wouldn't `start = 0` be same as not setting skip?

**Pull request merged**